### PR TITLE
[DEV-1627] Refactor tutorial

### PR DESCRIPTION
**Pull request by MarcoPonchia**
depends on: #841 

#### List of Changes

- Add Tutorial codec
- Add tutorials fetch function
- Refactor tutotials list page to manage cms tutorials
- Refactor tutotial page to manage cms tutorials

#### Motivation and Context

User can see tutorial from BE

#### How Has This Been Tested?

In order to test this PR you need to add a tutotial on Strapi (using this pr: https://github.com/pagopa/developer-portal/pull/841)

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by tommaso1 — APPROVED** [review]

**Review by tommaso1 — APPROVED** [review]

File: apps/nextjs-website/src/lib/api.ts [hunk 1]
```diff
@@ -95,10 +100,31 @@ export async function getQuickStartGuide(productSlug?: string) {
...
+}
+
+export async function getStrapiTutorial(
+  productSlug?: string,
+  productTutorialPage?: ReadonlyArray<string>
+) {
+  const tutorialSubPath = productTutorialPage?.join('/');
+  const tutorialPath = `/${productSlug}/tutorials/${tutorialSubPath}`;
+
+  const tutorialsFromCMS = await getTutorialsProps();
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/api.ts:113]: The function name is `getStrapiTutorial`, why this constant name refers to `CMS`?

File: apps/nextjs-website/src/lib/strapi/__tests__/tutorials.test.ts [hunk 2]
```diff
@@ -0,0 +1,321 @@
+import * as E from 'fp-ts/lib/Either';
+import { StrapiTutorialsCodec } from '@/lib/strapi/tutorial';
+
+const baseTutorialJson = {
+  id: 1,
+  attributes: {
+    title: 'title',
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/__tests__/tutorials.test.ts:7]: You could replace these texts and use the functions in `mock-content.helper.ts`

File: apps/nextjs-website/src/lib/strapi/tutorial.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import { NullToUndefinedCodec } from './codecs/NullToUndefinedCodec';
+import { BlocksContentCodec } from './codecs/BlocksContentCodec';
+import { PaginationCodec } from './codecs/PaginationCodec';
+import { ProductCodec } from './codecs/ProductCodec';
+import { RelatedLinksCodec } from './codecs/RelatedLinksCodec';
+
+const BannerLinkCodec = t.strict({
+  id: t.number,
+  title: t.union([NullToUndefinedCodec, t.string]),
+  body: t.union([NullToUndefinedCodec, BlocksContentCodec]),
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/tutorial.ts:16]: What about `icon` and `theme`?

```
type BannerLinkProps = {
  id: number;
  theme?: 'dark' | 'light';
  title?: string;
  body?: BlocksContent;
  icon: {
    readonly name: string;
    readonly alternativeText: string | null;
    readonly width: number;
    readonly height: number;
    readonly ext: string;
    readonly mime: string;
    readonly url: string;
  };
};
```

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/lib/strapi/tutorial.ts (hunk 3 above)
Comment (MarcoPonchia) [reply apps/nextjs-website/src/lib/strapi/tutorial.ts:16]: I know some attributes are added a next PR and about other missing attribute i thinnk is better to manage the refactor in a separeted task so we can align Strapi type and all FE occurencies

File: apps/nextjs-website/src/lib/strapi/__tests__/tutorials.test.ts (hunk 2 above)
Comment (MarcoPonchia) [reply apps/nextjs-website/src/lib/strapi/__tests__/tutorials.test.ts:7]: mock content helper it is an helper designed to be used in stories, here in tests i think is better to use raw json of http call

File: apps/nextjs-website/src/lib/strapi/tutorial.ts (hunk 3 above)
Comment (marcobottaro) [reply apps/nextjs-website/src/lib/strapi/tutorial.ts:16]: https://pagopa.atlassian.net/browse/DEV-1725

File: package-lock.json [hunk 4]
```diff
@@ -98,7 +98,7 @@
         "@testing-library/jest-dom": "^6.4.2",
         "@testing-library/react": "^14.1.2",
         "@types/jest": "^29.5.1",
-        "@types/node": "18.16.*",
+        "@types/node": "18.16.20",
```
Comment (marcobottaro) [inline package-lock.json:101]: I think you should remove this change

**Review by marcobottaro — CHANGES_REQUESTED** [review]

**Review by marcobottaro — APPROVED** [review]

File: apps/nextjs-website/src/app/[productSlug]/tutorials/[...productTutorialPage]/page.tsx [hunk 5]
```diff
@@ -64,7 +81,34 @@ const Page = async ({ params }: { params: Params }) => {
...
+      <ProductLayout
+        product={strapiTutorialProps.product}
+        path={strapiTutorialProps.path}
+        // bannerLinks={strapiTutorialProps.bannerLinks} // TODO: refactor bannerLinks
+      >
+        <Box mt={5}>
+          {strapiTutorialProps.title && (
+            <Abstract title={strapiTutorialProps.title} />
+          )}
+          <BlocksRendererClient content={strapiTutorialProps.content} />
```
Comment (marcobottaro) [inline apps/nextjs-website/src/app/[productSlug]/tutorials/[...productTutorialPage]/page.tsx:99]: I think we should add an `EContainer` around the `BlocksRendererClient`:
image

File: apps/nextjs-website/src/lib/strapi/tutorial.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+});
+
+export const TutorialCodec = t.strict({
+  id: t.number,
+  attributes: t.strict({
+    title: t.string,
+    slug: t.string,
+    content: BlocksContentCodec,
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/tutorial.ts:24]: Maybe we should refactor both BE and FE and replace this BlockContentCodec with Parts to manage code blocks and other parts
image

**Review by marcobottaro — CHANGES_REQUESTED** [review]

**Review by tommaso1 — APPROVED** [review]

File: apps/nextjs-website/src/lib/strapi/tutorial.ts (hunk 6 above)
Comment (marcobottaro) [reply apps/nextjs-website/src/lib/strapi/tutorial.ts:24]: I've inserted a new story about this activity in the project's backlog:
https://pagopa.atlassian.net/browse/DEV-1742

**Review by marcobottaro — APPROVED** [review]

**Review by tommaso1 — APPROVED** [review]

**Pull request merged**